vector/src/main/java/im/vector/app/VectorApplication.kt

```diff
@@ -77,7 +75,6 @@ import androidx.work.Configuration as WorkConfiguration
 @HiltAndroidApp
 class VectorApplication :
         Application(),
-        MatrixConfiguration.Provider,
```

Comment:
Maybe delete this interface? Or can it still be used by other application?

Reply:
it could be used by other clients of the SDK, if we agree that the Matrix singleton API should avoid querying the application then we could mark the interface a deprecate and schedule removing it? potentially with some migration steps~ 

what do you think?

Reply:
Yes I think we should remove this and the inner singleton to avoid mistakes! 
Regarding the process, we could indeed mark them as deprecated in a first place.